nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/eventhub/utils/AzureEventHubUtils.java

```diff
@@ -103,4 +111,42 @@ public static Map<String, String> getApplicationProperties(final Map<String,Obje
 
         return properties;
     }
+
+    /**
+     * Creates the {@link ProxyOptions proxy options}.
+     *
+     * @param propertyContext to supply Proxy configurations
+     * @return {@link ProxyOptions proxy options}, null if Proxy is not set
+     */
+    public static ProxyOptions getProxyOptions(final PropertyContext propertyContext) {
```

Comment:
Instead of returning a nullable `ProxyOptions`, what do you think about returning it in an `Optional` wrapper? That would indicate the contract more clearly.